src/oncall/scheduler/default.py

```diff
@@ -83,15 +83,19 @@ def get_busy_user_by_event_range(self, user_ids, team_id, events, cursor, table_
                        team_id)
         subscriptions = cursor.fetchall()
         team_check = ['team_id = %s']
+
+        cursor.execute("SELECT `id` FROM `role` WHERE `name` = 'vacation'")
+        vacation_role_id = cursor.fetchone()['id']
+
         query_params.append(team_id)
         for sub in subscriptions:
             team_check.append('(team_id = %s AND role_id = %s)')
             query_params += [sub['subscription_id'], sub['role_id']]
 
         query = '''
                 SELECT DISTINCT `user_id` FROM `%s`
-                WHERE `user_id` in %%s AND (%s) AND (%s)
-                ''' % (table_name, ' OR '.join(range_check), ' OR '.join(team_check))
+                WHERE `user_id` in %%s AND (%s) AND (%s or `role_id`= %s)
```

Comment:
You can avoid the separate query above like this:


    `role_id` = (SELECT `id` FROM `role` WHERE `name` = 'vacation')